checker/src/org/checkerframework/checker/index/lowerbound/LowerBoundAnnotatedTypeFactory.java

```diff
@@ -309,21 +309,39 @@ public Void visitMethodInvocation(MethodInvocationTree tree, AnnotatedTypeMirror
         }
 
         /**
+         * Looks up the minlen of a member select tree. The tree must be an access to a sequence
+         * length.
+         */
+        private Integer getMinLenFromTree(Tree tree) {
+            AnnotatedTypeMirror minLenType = getValueAnnotatedTypeFactory().getAnnotatedType(tree);
+            Long min = getValueAnnotatedTypeFactory().getMinimumIntegralValue(minLenType);
+            if (min == null) {
+                return null;
+            }
+            if (min < 0 || min > Integer.MAX_VALUE) {
+                min = 0L;
+            }
+            return min.intValue();
+        }
+
+        /**
          * Looks up the minlen of a member select tree. Returns null if the tree doesn't represent
          * an array's length field.
          */
         private Integer getMinLenFromMemberSelectTree(MemberSelectTree tree) {
             if (TreeUtils.isArrayLengthAccess(tree)) {
-                AnnotatedTypeMirror minLenType =
-                        getValueAnnotatedTypeFactory().getAnnotatedType(tree);
-                Long min = getValueAnnotatedTypeFactory().getMinimumIntegralValue(minLenType);
-                if (min == null) {
-                    return null;
-                }
-                if (min < 0 || min > Integer.MAX_VALUE) {
-                    min = 0L;
-                }
-                return min.intValue();
+                return getMinLenFromTree(tree);
+            }
+            return null;
+        }
+
+        /**
+         * Looks up the minlen of a mehtod invocation tree. Returns null if the tree doesn't
+         * represent an string length method.
+         */
+        private Integer getMinLenFromMethodInvocationTree(MethodInvocationTree tree) {
```

Comment:
This would mean the method would have 4 parameters `(MethodInvocationTree, ValueAnnotatedTypeFactory, IndexMethodIdentifier, ProcessingEnvironment)`. I think it would be less readable, given how short the method is. Is it worth it?